### DISCUSSION TITLE
Not possible to set an `object` with `additionalProperties`

### DISCUSSION THREAD

**Discussion by WesleyYue** [fastapi/fastapi#10623] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()

class Foo(BaseModel):
    bars: Optional[Dict[str, Any]] = None

@app.get("/")
def read_root(baz: Foo):
    return {"Hello": "World"}
```

### Description

As far as I can tell, it's not possible to output a OpenAPI schema that has the following schema for `bars`:
```
type: object
additionalProperties: true
```

In the example code, there is no way to set `bars` to have `extra = "allow"`, since RootModels from Pydantic does not allow `extra` to be set to anything but None.

I need this because I have a downstream [OpenAPI codegen tool](https://openapi-ts.pages.dev/advanced/#:~:text=Be%20specific%20in%20your%20schema) that relies on having `additionalProperties`. Am I missing something obvious or is there no way to do this in FastAPI?

### Operating System

macOS

### FastAPI Version

0.104.1

### Pydantic Version

2.4.2

### Python Version

Python 3.11.5

**Comment by blakjak44** [comment, marked as answer]
Figured I'd post an answer since I had the exact same use case. You can use the `json_schema_extra` argument to `Field`:

```python
from pydantic import BaseModel, Field

class Foo(BaseModel):
    bars: dict = Field(
        default={},
        json_schema_extra={
            'additionalProperties': True,
        }
    )
```
